### DISCUSSION TITLE
Naming convention - azd should follow recommended abbreviations for Azure resource types

### DISCUSSION THREAD

**Discussion by cbezenco** [Azure/azure-dev#127] [Ideas]
I know naming conventions are subjective but to be consistent, I believe azd should follow MIcrosoft Azure best practices for resource abbreviations naming convention whenever possible : https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations

**Comment by savannahostrowski** [comment]
I believe we already are following the guidance in this doc. @jongio can you confirm?

**Reply by jongio** [reply]
Yes, we are following the guidelines from that doc.  Please lmk if you see any that do not.

**Reply by cbezenco** [reply]
There is a small mismatch. You are following the convention for the template resources but not for the resource group name. 
You use ''{name}-rg''  vs ''rg-{name}'' in the main. Bicep infra. 

resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
  name: '${name}-rg'
  location: location
  tags: tags
}

**Reply by jongio** [reply]
Thanks, yes, I'm working on a solution to share consistent abbreviations across all templates, including defaulting to "rg-" instead of "-rg".

**Reply by jongio** [reply]
Here's the PR that will bring consistency to abbreviations used and change resource group to a prefix.  You are welcome to try it out and provide feedback.  Links to the CLI and updated templates are in comments in the PR.

https://github.com/Azure/azure-dev/pull/140

**Reply by jongio** [reply]
The fix for this has been merged and will be available with our next release https://github.com/Azure/azure-dev/pull/140
